litgpt/finetune/adapter.py

```diff
@@ -209,9 +210,13 @@ def fit(
                 time=t1 - total_t0, batches=iter_num, samples=iter_num * train.micro_batch_size, lengths=total_lengths
             )
             throughput.compute_and_log(step=iter_num)
+            val_loss_str = 'n/a' if val_loss is None else f"{val_loss:.3f}"
```

Comment:
Should we cache the `val_loss` string? Validation doesn't run often so we could optimize this a bit by reusing the string that we already computed.

`val_loss` could be initialized to `n/a` and then we only recompute it here if it's a tensor

```suggestion
            if isinstance(val_loss, torch.Tensor):
                val_loss = f"{val_loss:.3f}"
```

Reply:
I profiled 3 prints of the same cuda tensor. PyTorch seems to be smart already and the time it takes for the second and third prints is much lower (see highlighted section compared to the left side). But it still triggers stream synchronization that we could simply avoid

![image](https://github.com/Lightning-AI/litgpt/assets/11067892/3a0450e6-e16a-441f-878f-1e9808198fc7)

Reply:
Wow nice. Yeah, let's use your suggestion. I added it to all the scripts. Note that in the pretraining script, the `val_loss` is a `float` not a `torch.Tensor`.

Reply:
Are you sure about that? The validaiton function is annotated as returning a Tensor

Reply:
Ah thanks. I missed that. In that case my suggestion is not relevant for pretrain.py. Looks good anyways